src/components/Details.js

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect } from 'react';
+import { useParams, NavLink } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import { dataList } from '../redux/home/home';
+import './Details.css';
+
+function Details() {
+  const dispatch = useDispatch();
+  const parameters = useParams();
+  const { name } = parameters;
+  const listOfData = useSelector((state) => state.home);
+  useEffect(() => {
+    if (!listOfData.length) {
+      dispatch(dataList());
+    }
+  }, [dispatch, listOfData.length]);
+  const filteredList = listOfData.filter((item) => item.name === name);
```

Comment:
- [ ] [Optional and important] this could be managed in your state, this ensures that your global state is your only source of truth :+1: